sympy/core/sympify.py

```diff
@@ -252,6 +252,24 @@ def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
         else:
             return a
 
+    #Support for basic numpy datatypes
+    if type(a).__module__ == 'numpy':
+        import numpy as np
+        if np.isscalar(a):
+            if not isinstance(a, np.floating):
+                return sympify(np.asscalar(a))
+            else:
+                try:
+                    from sympy.core import N
+                    from mpmath.libmp.libmpf import repr_dps
+                    prec = np.finfo(a).nmant
+                    a = str(list(np.reshape(np.asarray(a),
+                                            (1, np.size(a)))[0]))[1:-1]
+                    return N(a, repr_dps(prec)-1)
```

Comment:
This looks good. It's probably better to use `Float` instead of `N` here. `N` calls `sympify`, which creates a Float, then evalf, which creates another Float. So it's better to use `Float` here and just create one Float. 

I wish the Float constructor had support for setting the binary precision directly, instead of just the base-10 precision, but that is a separate issue.